### PR TITLE
test (datafusion): add test for table provider creation

### PR DESCRIPTION
**Pull request by a-agmon**
Adding tests for #650  - to test table provider creation from a static table

### CODE REVIEW

File: crates/integrations/datafusion/src/table.rs [hunk 1]
```diff
@@ -110,3 +110,43 @@ impl TableProvider for IcebergTableProvider {
...
+        let static_identifier = TableIdent::from_strs(["static_ns", "static_table"]).unwrap();
+        let static_table =
+            StaticTable::from_metadata_file(&metadata_file_path, static_identifier, file_io)
+                .await
+                .unwrap();
+        static_table.into_table()
+    }
+
+    #[tokio::test]
+    async fn test_try_new_from_table() {
```
Comment (FANNG1) [inline crates/integrations/datafusion/src/table.rs:142]: could you use datafusion to test the Iceberg table schema?

Comment (a-agmon) [reply crates/integrations/datafusion/src/table.rs:142]: you mean adding something like this:
```rust
let ctx = SessionContext::new();
ctx.register_table("mytable", Arc::new(table_provider))
    .unwrap();
let df = ctx.sql("SELECT * FROM mytable").await.unwrap();
```

or running an actual query using it? that would require a real table (the dummy metadata I used is not really pointing at a real location)

Comment (a-agmon) [reply crates/integrations/datafusion/src/table.rs:144]: Sure. Done. 
Thanks @FANNG1

**Review by Xuanwo — APPROVED** [review]
Great, thanks!

**Comment by Xuanwo** [comment]
Thank you @a-agmon to build this and @FANNG1 for the review!

**Pull request merged**